Client/game_sa/CFxSA.cpp

```diff
@@ -237,3 +241,42 @@ void CFxSA::TriggerFootSplash(CVector& vecPosition)
         call    dwFunc
     }
 }
+
+bool CFxSA::IsShadowsLimitReached()
+{
+    short shadows = *(short*)0xC403DC;
+    return shadows >= 48;
+}
+
+bool CFxSA::AddShadow(eShadowType shadowType, CVector& vecPosition, CVector2D& vecOffset1, CVector2D& vecOffset2, SColor color, float fZDistance,
+                      bool bDrawOnWater, bool bDrawOnBuildings)
+{
+    if (IsShadowsLimitReached())
+        return false;
+
+    RwTexture* pTexture = nullptr;
+    char       type = 1;
+    if (shadowType == SHADOW_NONE)
+    {
+        pTexture = EmptyTexture;
+    }

```

Comment:
I think making `SHADOW_NONE` actually mean "empty texture" is confusing and should be changed.